engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1702,6 +1705,18 @@ protected boolean checkVmOnHost(final VirtualMachine vm, final long hostId) thro
             }
         }
 
+        UserVmVO userVm = _userVmDao.findById(vm.getId());
+        if (userVm != null) {
+            List<VMSnapshotVO> vmSnapshots = _vmSnapshotDao.findByVm(vm.getId());
+            RestoreVMSnapshotCommand command = _vmSnapshotMgr.createRestoreCommand(userVm, vmSnapshots);
+            if (command != null) {
+                RestoreVMSnapshotAnswer restoreVMSnapshotAnswer = (RestoreVMSnapshotAnswer) _agentMgr.send(hostId, command);
+                if (restoreVMSnapshotAnswer == null || !restoreVMSnapshotAnswer.getResult()) {
+                    s_logger.warn("Unable to restore the vm snapshot from image file after migration to the VM: " + restoreVMSnapshotAnswer.getDetails());
```

Comment:
What is the migration (type?) mentioned in this log message? Can you please elaborate this a bit?